Client/Frontend/Browser/Playlist/Controllers/PlaylistNewFolderView.swift

```diff
@@ -0,0 +1,198 @@
+// Copyright 2022 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import SwiftUI
+import Combine
+import Data
+import CoreData
+import Shared
+import BraveShared
+
+private struct PlaylistFolderImage: View {
+    static let cornerRadius = 10.0
+    private static let favIconSize = 16.0
+    private var thumbnailLoader: ImageLoader
+    private var favIconLoader: ImageLoader
+    
+    private var title: String?
+    @State private var thumbnail = UIImage()
+    @State private var favIcon = UIImage()
+    
+    init(item: PlaylistItem) {
+        title = item.name
+        thumbnailLoader = ImageLoader(thumbnail: item)
+        favIconLoader = ImageLoader(favIcon: item)
+    }
+    
+    var body: some View {
+        Image(uiImage: thumbnail)
+            .resizable()
+            .aspectRatio(contentMode: .fit)
+            .background(Color.black)
+            .overlay(tint)
+            .overlay(VStack(alignment: .leading) {
+                Image(uiImage: favIcon)
+                    .resizable()
+                    .aspectRatio(1.0, contentMode: .fit)
+                    .frame(width: PlaylistFolderImage.favIconSize,
+                           height: PlaylistFolderImage.favIconSize)
+                    .clipShape(RoundedRectangle(cornerRadius: 2.5, style: .continuous))
+                
+                Spacer()
+                
+                Text(title ?? "")
+                    .font(.footnote.weight(.semibold))
+                    .lineLimit(2)
+                    .foregroundColor(.white)
+            }.padding(8.0), alignment: .topLeading)
+            .clipShape(RoundedRectangle(cornerRadius: PlaylistFolderImage.cornerRadius, style: .continuous))
+            .onReceive(thumbnailLoader.$image) {
+                self.thumbnail = $0 ?? UIImage()
+            }
+            .onReceive(favIconLoader.$image) {
+                self.favIcon = $0 ?? UIImage()
+            }
+    }
+    
+    private var tint: some View {
+        EmptyView()
+        .frame(maxWidth: .infinity, maxHeight: .infinity)
+        .background(LinearGradient(colors: [.clear, .black],
+                                   startPoint: .top,
+                                   endPoint: .bottom))
+        .clipShape(RoundedRectangle(cornerRadius: PlaylistFolderImage.cornerRadius, style: .continuous))
+    }
```

Comment:
Not needed, you can use do 

```swift
.overlay(
  LinearGradient(...)
)
```

1. LinearGradient is greedy, it will already take up all space given (which inside overlay is the size of the view)
2. The `clipShape` is applied to the entire view so it doesn't need to be clipped again